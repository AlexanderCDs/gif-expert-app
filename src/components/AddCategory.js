/**
 * @author: alexandercd
 * @date: 22/septiembre/2021
 * @description: 
**/
import React, {useState} from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories, placeholder }) => {  
    const [inputValue, setInputValue] = useState(''); 
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories(c => [inputValue, ...c]);
            setInputValue('');
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <input 
                placeholder={placeholder}
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            /> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory;