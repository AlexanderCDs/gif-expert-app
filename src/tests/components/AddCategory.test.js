/**
 * @author: alexandercd
 * @date: 24/septiembre/2021
 * @description: 
**/
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Test <AddCategory />', () => {
    const value = 'Hola, Goku';  
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories}/>);
    });
    
    test('should show <AddCategory /> success', () => {   
        expect( wrapper ).toMatchSnapshot();
    }); 

    test('should change text into input', () => {
        const input = wrapper.find('input'); 
        input.simulate('change', {target: {value}});
        const changeValue = wrapper.find('input'); 
        expect( changeValue.prop('value').trim() ).toBe( value );
    });
    
    test('should not submit', () => {
        const from = wrapper.find('form');
        from.simulate('submit', {preventDefault(){}});
        expect( setCategories ).not.toHaveBeenCalled();
    });
    
    test('should call setCategories and clean input', () => {
        wrapper.find('input').simulate('change', {target: {value}}); 
        wrapper.find('form').simulate('submit', {preventDefault(){}}); 
        
        expect( setCategories ).toHaveBeenCalled(); 
        expect( setCategories ).toHaveBeenCalledTimes( 1 );
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value').trim() ).toBe( '' );
    });
    
});
