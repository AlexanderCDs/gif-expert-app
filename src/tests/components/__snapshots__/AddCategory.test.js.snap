// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory /> should show <AddCategory /> success 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
